File: WebApi/ClientApp/src/app/eshop/models/product.ts
import { Category } from "./category";
import { BaseType } from "./baseType";
import { Discount } from "./discount";

export interface Product extends BaseType {
  name: string,
  description: string,
  picruteUrl: string,
  quantity: number,
  category: Category,
  categoryId: number,
  price: number,
  discount: Discount,
  discountId: number
}

File: Data/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Data.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; }

    public string Description { get; set; }

    public string? PictureUrl { get; set; }

    public int Quantity { get; set; }

    public virtual Category Category { get; set; }

    public int CategoryId { get; set; }

    [Column(TypeName = "decimal(7, 2)")]
    public decimal Price { get; set; }

    public virtual Discount Discount { get; set; }

    public int DiscountId { get; set; }
}